.github/workflows/release.yaml

```
@@ -0,0 +1,182 @@
+#
+# Copyright Red Hat
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+'
+
+env:
+  OPERATOR_SDK_DL_URL: https://github.com/operator-framework/operator-sdk/releases/download/v1.28.0
+  OPERATOR_SDK_CLI: ./operator-sdk
+
+jobs:
+  version-check:
+    runs-on: ubuntu-latest
+    env:
+      CACHED_CSV_VERSION: .cache/csv_version.txt
+      CACHED_CSV_CONTAINER_IMAGE_TAG: .cache/csv_container_image_tag.txt
+      CACHED_CSV_NAME_TAG: .cache/csv_name_tag.txt
+      CACHED_BUNDLE_VERSION: .cache/bundle_version.txt
+      CACHED_BUNDLE_CONTAINER_IMAGE_TAG: .cache/bundle_container_image_tag.txt
+      CACHED_BUNDLE_NAME_TAG: .cache/bundle_name_tag.txt
+      CACHED_MANAGER_IMAGE_TAG: .cache/manager_image_tag.txt
+    steps:
+    - name: Checkout registry-operator source code
+      uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+    - name: Download yq
+      run: curl -sL https://github.com/mikefarah/yq/releases/download/v4.34.2/yq_linux_amd64 -o yq && chmod +x yq
+    - name: Cache version tags
+      env:
+        YQ_CLI: ./yq
+      run: mkdir -p .cache && bash .ci/cache_version_tags.sh
+    - name: Check if VERSION and version tags match release tag
+      run: bash check_version.sh ${{ github.ref_name }} true
+  k8s-1-26-integration-test:
+    needs: version-check
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout registry-operator source code
+      uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+    - name: Setup Go environment
+      uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+      with:
+        go-version: 1.19
```

How about using the [`go-version-file`](https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file) field instead? The `go.mod` will then be the source of truth, and it will be easier to manage if we bump to a new version of Go.

```suggestion
        go-version-file: 'go.mod'
```

---

resolved in https://github.com/devfile/registry-operator/pull/56/commits/cb65ba2693e47213120e8871d277260176fb50fb & https://github.com/devfile/registry-operator/pull/56/commits/c815d922f0a1413cb4d3354bcf055a5c084dd61a